crypto/asn1/a_gentm.c

```diff
@@ -100,7 +119,7 @@ int asn1_generalizedtime_to_tm(struct tm *tm, const ASN1_GENERALIZEDTIME *d)
 
     if (a[o] == 'Z')
         o++;
-    else if ((a[o] == '+') || (a[o] == '-')) {
+    else if (!strict && ((a[o] == '+') || (a[o] == '-'))) {
```

Comment:
There doesn't seem to be a check that the Z is actually present. I think it currently accepts any character as the last character.